apiserver/common/networkingcommon/networkconfigapi.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// The networkconfigapi package implements the network config parts
+// common to machiner and provisioner interface
+
+package networkingcommon
+
+import (
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/state"
+	"github.com/juju/juju/state/stateenvirons"
+)
+
+type NetworkConfigAPI struct {
+	st           *state.State
+	getCanModify common.GetAuthFunc
+	getAuthFunc  common.GetAuthFunc
+}
+
+func NewNetworkConfigAPI(st *state.State, getCanModify common.GetAuthFunc) *NetworkConfigAPI {
+	return &NetworkConfigAPI{
+		st:           st,
+		getCanModify: getCanModify,
+	}
+}
+
+func (api *NetworkConfigAPI) getMachine(tag names.Tag) (*state.Machine, error) {
+	entity, err := api.st.FindEntity(tag)
+	if err != nil {
+		return nil, err
+	}
+	return entity.(*state.Machine), nil
+}
+
+func (api *NetworkConfigAPI) getOneMachineProviderNetworkConfig(m *state.Machine) ([]params.NetworkConfig, error) {
+	instId, err := m.InstanceId()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	netEnviron, err := NetworkingEnvironFromModelConfig(
+		stateenvirons.EnvironConfigGetter{api.st},
+	)
```

Comment:
ah, this is probably something else that needs more of State.
Its a bit of a layering violation, but not one that you introduced. I wonder if we could pass in the EnvironConfigGetter instead of directly injecting state. It still isn't ideal, but it might be a little bit better.